outport/outport.go

```diff
@@ -1,27 +1,39 @@
 package outport
 
 import (
+	"fmt"
 	"sync"
+	"time"
 
 	"github.com/ElrondNetwork/elrond-go-core/core/check"
 	"github.com/ElrondNetwork/elrond-go-core/data"
 	"github.com/ElrondNetwork/elrond-go-core/data/indexer"
 	logger "github.com/ElrondNetwork/elrond-go-logger"
 )
 
+var log = logger.GetOrCreate("outport")
+
+const minimumRetrialInterval = time.Millisecond * 10
+
 type outport struct {
-	mutex   sync.RWMutex
-	drivers []Driver
+	mutex           sync.RWMutex
+	drivers         []Driver
+	retrialInterval time.Duration
+	chanClose       chan struct{}
 }
 
-var log = logger.GetOrCreate("outport")
-
 // NewOutport will create a new instance of proxy
-func NewOutport() *outport {
-	return &outport{
-		drivers: make([]Driver, 0),
-		mutex:   sync.RWMutex{},
+func NewOutport(retrialInterval time.Duration) (*outport, error) {
+	if retrialInterval < minimumRetrialInterval {
+		return nil, fmt.Errorf("%w, provided: %d, minimum: %d", ErrInvalidRetrialInterval, retrialInterval, minimumRetrialInterval)
```

Comment:
:+1:  
we can avoid high(100%) cpu load if flag is missing

Reply:
exactly. We should always have minimums when doing endless fors.